Mlem/Extensions/Swipey Actions.swift

```diff
@@ -202,63 +158,72 @@ struct SwipeyView: ViewModifier {
         // disables links from highlighting when tapped
         .buttonStyle(EmptyButtonStyle())
     }
+    
+    private func draggingDidEnd() {
+        // TODO: instant upvote feedback (waiting on backend)
+        if prevDragPosition < -1 * AppConstants.longSwipeDragMin {
+            Task(priority: .userInitiated) {
+                let action = secondaryTrailingAction ?? primaryTrailingAction
+                await action?.action()
+            }
+        } else if prevDragPosition < -1 * AppConstants.shortSwipeDragMin {
+            Task(priority: .userInitiated) {
+                await primaryTrailingAction?.action()
+            }
+        } else if prevDragPosition > AppConstants.longSwipeDragMin {
+            Task(priority: .userInitiated) {
+                let action = secondaryLeadingAction ?? primaryLeadingAction
+                await action?.action()
+            }
+        } else if prevDragPosition > AppConstants.shortSwipeDragMin {
+            Task(priority: .userInitiated) {
+                await primaryLeadingAction?.action()
+            }
+        }
+        
+        reset()
+    }
+    
+    private func reset() {
+        withAnimation(.interactiveSpring()) {
+            dragPosition = .zero
+            leadingSwipeSymbol = primaryLeadingAction?.symbol.emptyName
+            trailingSwipeSymbol = primaryTrailingAction?.symbol.emptyName
+            dragBackground = .systemBackground
+        }
+    }
+    
+    private func shouldRespondToDragPosition(_ position: CGFloat) -> Bool {
+        if position > 0, primaryLeadingAction == nil {
+            return false
+        }
+        
+        if position < 0, primaryTrailingAction == nil {
+            return false
+        }
+        
+        return true
+    }
```

Comment:
Really like this function, it's going to make it easier to fine-tune drag detection too